src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -613,33 +609,56 @@ bool LibraryCallKit::inline_vector_shuffle_iota() {
   int num_elem = vlen->get_con();
   BasicType elem_bt = T_BYTE;
 
-  if (!arch_supports_vector(VectorNode::replicate_opcode(elem_bt), num_elem, elem_bt, VecMaskNotUsed)) {
-    return false;
+  bool effective_indices_in_range = false;
+  if (start_val->is_con() && step_val->is_con()) {
+    int effective_min_index = start_val->get_con();
+    int effective_max_index = start_val->get_con() + step_val->get_con() * (num_elem - 1);
+    effective_indices_in_range = effective_max_index >= effective_min_index && effective_min_index >= -128 && effective_max_index <= 127;
   }
-  if (!arch_supports_vector(Op_AddVB, num_elem, elem_bt, VecMaskNotUsed)) {
-    return false;
-  }
-  if (!arch_supports_vector(Op_AndV, num_elem, elem_bt, VecMaskNotUsed)) {
+
+  if (!do_wrap && !effective_indices_in_range) {
+    // FIXME: disable instrinsification for unwrapped shuffle iota
```

Comment:
Please don't add FIXME's to new code or at least file a follow-up RFE and reference it here.

Reply:
> Please don't add FIXME's to new code or at least file a follow-up RFE and reference it here.

JDK-8311305 filed to address this in follow-up RFE.